examples/bpi-monolith-test-suite/bpi/components/workgroup/workgroup.ts

```diff
@@ -6,16 +6,24 @@ export class Workgroup {
      name: string;
      participants: BpiSubject[] = [];
      worksteps: Workstep[] = [];
+     workstepids: Set<string> = new Set("");
 
     constructor(name: string, id: string, owner: BpiSubject, worksteps: Workstep[],) {
-        this.name = name;
         this.id = id;
+        this.name = name;
         this.worksteps = worksteps;
         this.participants.push(owner);
     }
 
      addWorkstep(workstep: Workstep) {
-          this.worksteps.push(workstep);
+
+          //push workstep if ID is not yet in use
+          if(!(this.workstepids.has(workstep.id))) {
```

Comment:
i dont think we need parenthesis after `!`

also, a bit unclear here, why multiple worksteps can have same id? does this mean that addWorkstep can be called multiple times for same workstep? how much worksteps there can be, if it's usually small number maybe adding set for this to optimize search might not be worth and you can just use getWorkstepById? maybe we can rename function as `addWorkstepIfNotExists`, and simplify it like:

if (this.workstepIds.has(workstep.id)) {
return;
}

this.worksteps.push(workstep);
this.workstepids.add(workstep.id);

Reply:
The reason for these changes is to prevent someone from reusing a previously used ID in a workgroup. Per R217 this would be in violation of the standard, so my changes here is to make it impossible for an ID to be reused, by tracking the IDs in current use in the set. I considered using getWorkstepById, but I worried about search time complexity. In this small test suite it is fast either way, and not much of a concern, but I worried that the scalability is lost, and wanted a solution that can be used for any implementation of any size. What are your thoughts on this?